### PR TITLE
Development of the tn4ml library

### PR DESCRIPTION
**Pull request by gabrieledangeli**
Here's a brief description of my changes:

- `setup.py`: Fixed installation requirements (tested, should be good now).

- `embeggings.py`: Implemented the `PatchAmplitudeEmbedding` class (training suggests not using this embedding). Added a ceiling function to ensure the correct number of address qubits in the `AmplitudeEmbedding` class.

- `metrics.py`: Implemented the `CombinedOptaxLoss` class to handle combined Log Softmax Cross Entropy loss with L2 Regularization.

- `model.py`: Fixed a small bug in the `predict` function when checking the correct number of tensors in the MPS.

- `utils.py`: Fixed the `pad_images` function.

### CODE REVIEW

File: tn4ml/util.py
Comment (emapuljak) [inline tn4ml/util.py:1]: This would mean that image is square-like?

Comment (gabrieledangeli) [reply tn4ml/util.py:1]: The problem was that the padding zeros were added only to one side of the image and not uniformly across its border

Comment (emapuljak) [reply tn4ml/util.py:1]: Ah okay, right!
